docs/maintenance.md

```diff
@@ -0,0 +1,81 @@
+Maintenance guide
+=================
+
+This is the maintenance guide for pvc-autoresizer.
+
+How to upgrade supported Kubernetes version
+-------------------------------------------
+
+pvc-autoresizer depends on some Kubernetes repositories like `k8s.io/client-go` and should support 3 consecutive Kubernets versions at a time.
+Here is the guide for how to upgrade the supported versions.
+Issues and PRs related to the last upgrade task also help you understand how to upgrade the supported versions,
+so checking them together with this guide is recommended when you do this task.
+
+### Check release notes
+
+First of all, we should have a look at the release notes in the order below.
+
+1. Kubernetes
+    - Choose the next version and check the [release note](https://kubernetes.io/docs/setup/release/notes/). e.g. 1.17, 1.18, 1.19 -> 1.18, 1.19, 1.20
+    - Read the [release note](https://github.com/kubernetes-sigs/controller-runtime/releases), and check which version is compatible with the Kubernetes versions.
+    - Read the [kubebuilder go.mod](https://github.com/kubernetes-sigs/kubebuilder/blob/master/go.mod), and check the controller-tools version corresponding to controller-runtime.
+2. TopoLVM
+    - Choose the [TopoLVM](https://github.com/topolvm/topolvm/releases) version that supported target Kubernetes version.
+3. Depending tools
+    - They does not depend on other software, use latest versions.
+      - [helm](https://github.com/helm/helm/releases)
+      - [helm-docs](github.com/norwoodj/helm-docs/releases)
+      - [kube-prometheus](https://github.com/prometheus-operator/kube-prometheus/releases)
+4. Depending modules
+  - Read [kubernetes go.mod](https://github.com/kubernetes/kubernetes/blob/master/go.mod), and update the `prometheus/*` modules.
+  - Read [csi-test go.mod](https://github.com/kubernetes-csi/csi-test/blob/master/go.mod), and update the `ginkgo` and `gomega` modules.
+
+Please write down to the Github issue of this task what kinds of changes we find in the release note and what we are going to do and NOT going to do to address the changes.
+The format is up to you, but this is very important to keep track of what changes are made in this task, so please do not forget to do it.
+
+Basically, we should pay attention to breaking changes and security fixes first.
+If we find some interesting features added in new versions, please consider if we are going to use them or not and make a GitHub issue to incorporate them after the upgrading task is done.
+
+### Update written versions
+
+Once we decide the versions we are going to upgrade, we should update the versions written in the following files manually.
+
+- `README.md`: Documentation which indicates what versions are supported by pvc-autoresizer
+- `Makefile`: Makefile for running envtest
+- `e2e/Makefile`: Makefile for running e2e tests
+
+`git grep 1.18`, `git grep image:`, and `git grep -i VERSION` might help us avoid overlooking necessary changes.
```

Comment:
```suggestion
`git grep <the kubernetes version which support will be dropped>, `git grep image:`, and `git grep -i VERSION` might help to avoid overlooking necessary changes.
```